src/app/accounts/accept-organization.component.ts

```diff
@@ -51,8 +60,39 @@ export class AcceptOrganizationComponent implements OnInit {
                     const request = new OrganizationUserAcceptRequest();
                     request.token = qParams.token;
                     try {
-                        this.actionPromise = this.apiService.postOrganizationUserAccept(qParams.organizationId,
-                            qParams.organizationUserId, request);
+                        if (await this.performResetPasswordAutoEnroll(qParams)) {
+                            this.actionPromise = this.apiService.postOrganizationUserAccept(qParams.organizationId,
+                                qParams.organizationUserId, request).then(() => {
+                                    // Retrieve Public Key
+                                    return this.apiService.getOrganizationKeys(qParams.organizationId);
+                                }).then(async response => {
+                                    let publicKey = null;
+
+                                    if (response != null) {
+                                        publicKey = Utils.fromB64ToArray(response.publicKey);
+                                    } else {
+                                        throw new Error('Get Organization Keys response is null');
+                                    }
+
+                                    // RSA Encrypt user's encKey.key with organization public key
+                                    const encKey = await this.cryptoService.getEncKey();
+                                    const encryptedKey = await this.cryptoService.rsaEncrypt(encKey.key, publicKey.buffer);
+
+                                    // Create request and execute enrollment
+                                    const resetRequest = new OrganizationUserResetPasswordEnrollmentRequest();
+                                    resetRequest.resetPasswordKey = encryptedKey.encryptedString;
```

Comment:
These operations seems to be used in the org settings component as well, could we extract them to prevent duplicate code?

Reply:
@Hinton Does it make sense here to maybe make a new service at the `jslib` layer to house these re-used encryption steps?

Reply:
I don't believe we need to do these org steps in the other clients?

Reply:
Nope. Good point.